provider/datagen/src/transform/cldr/transforms/mod.rs

```diff
@@ -0,0 +1,255 @@
+                .map_err(|e| {
+                    DataError::custom("transliterator parsing failed").with_debug_context(&e)
+                })?;
+        let transliterator = if is_forwards {
+            // the parser guarantees we receive this
+            #[allow(clippy::unwrap_used)]
+            forwards.unwrap()
+        } else {
+            // the parser guarantees we receive this
+            #[allow(clippy::unwrap_used)]
+            backwards.unwrap()
+        };
+
+        Ok(DataResponse {
+            metadata: Default::default(),
+            payload: Some(DataPayload::from_owned(transliterator)),
+        })
+    }
+}
+
+impl IterableDataProvider<TransliteratorRulesV1Marker> for crate::DatagenProvider {
+    fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
+        let mut locales = Vec::new();
+        for transform in self.cldr()?.transforms().list_transforms()? {
+            let metadata = self
+                .cldr()?
+                .transforms()
+                .read_and_parse_metadata(&transform)?;
+            let (forwards, backwards) = internal_ids_from_metadata(metadata);
+            if let Some(forwards) = forwards {
+                locales.push(format!("und+{forwards}").parse()?);
+            }
+            if let Some(backwards) = backwards {
+                locales.push(format!("und+{backwards}").parse()?);
+            }
+        }
+        Ok(locales)
+    }
+}
+
+/// Get the internal ICU4X ID for this transliterator from CLDR metadata.
+///
+/// Returns (forwards, backwards) internal IDs if the corresponding direction is supported according
+/// to the metadata.
+fn internal_ids_from_metadata(metadata: &transforms::Resource) -> (Option<String>, Option<String>) {
+    let forwards = if matches!(
+        metadata.direction,
+        transforms::Direction::Forward | transforms::Direction::Both
+    ) {
+        Some(internal_id_from_parts(
+            &metadata.alias,
+            &metadata.source,
+            &metadata.target,
+            metadata.variant.as_deref(),
+        ))
+    } else {
+        None
+    };
+    let backwards = if matches!(
+        metadata.direction,
+        transforms::Direction::Backward | transforms::Direction::Both
+    ) {
+        Some(internal_id_from_parts(
+            &metadata.backward_alias,
+            &metadata.target,
+            &metadata.source,
+            metadata.variant.as_deref(),
+        ))
+    } else {
+        None
+    };
+
+    (forwards, backwards)
+}
+
+fn internal_id_from_parts(
+    aliases: &[String],
+    source: &str,
+    target: &str,
+    variant: Option<&str>,
+) -> String {
+    find_bcp47_in_list(aliases).unwrap_or_else(|| {
+        icu_transliterator_parser::legacy_id_to_internal_id(source, target, variant)
+    })
+}
+
+fn find_bcp47_in_list(list: &[String]) -> Option<String> {
+    for item in list {
+        if item.contains("-t-") {
```

Comment:
might be better to parse to `Locale`.

Reply:
Parsing aliases as an enum over Locale (for -t-) and String (for legacy IDs) now